### PR TITLE
Adding Mastercard Encryption Support

### PR DESCRIPTION
**Pull request by danny-gallagher**
Added support for Mastercard Encryption to make consistent with other libraries.

### CODE REVIEW

**Comment by phongnguyenthanh14** [comment]
Hi @danny-gallagher when you merge this PR?

**Comment by danny-gallagher** [comment]
> Hi @danny-gallagher when you merge this PR?

Hi @phongnguyenthanh14, we just need to do some testing internally and review this PR before merging it. Apologies for the delay.

File: utils/encoding_utils.go [hunk 1]
```diff
@@ -14,3 +15,28 @@ func Base64UrlDecode(data string) []byte {
...
+		return Base64UrlEncode(data)
+	} else {
+		return HexUrlEncode(data)
+	}
+}
+
+func DecodeData(data string, encodingType string) []byte {
+	if encodingType == "BASE64" {
+		return Base64UrlDecode(data)
+	} else {
```
Comment (karen-avetisyan-mc) [inline utils/encoding_utils.go:39]: can you please make the encoding type selection specific, and add error handing if non supported type is selected

File: utils/string_utils.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package utils
+
+func IsNullOrEmpty(str string) bool {
+	return str == ""
```
Comment (karen-avetisyan-mc) [inline utils/string_utils.go]: If the code shall catch all empty characters, including"    " cases - then it is better to have the one below.
s= strings.Trim(s, " ") == ""
return s==""

**Review by karen-avetisyan-mc — COMMENTED** [review]
Some minor comments from me

File: utils/string_utils.go (hunk 2 above)
Comment (danny-gallagher) [reply utils/string_utils.go]: Done

File: utils/encoding_utils.go (hunk 1 above)
Comment (danny-gallagher) [reply utils/encoding_utils.go:39]: Added a check to 'field_level_encryption_config_builder' so we know at this stage that the encoding type will be either BASE64 or HEX

**Review by karen-avetisyan-mc — APPROVED** [review]
LGTM

**Pull request merged**